### ISSUE TITLE
Exception during mining

### ISSUE THREAD

**Issue by ckrause** [loda-lang/loda-cpp#288]
```
2023-10-04 19:28:18|INFO |Starting LODA v23.9.30. See https://loda-lang.org/
2023-10-04 19:28:18|INFO |Found environment variable: PROJECT_DIR=.\
2023-10-04 19:28:18|INFO |Loading init data from file: .\init_data.xml
2023-10-04 19:28:18|INFO |Platform: windows, system memory: 8103 MiB
2023-10-04 19:28:18|INFO |User name: ..., host ID: ...
2023-10-04 19:28:18|INFO |Using LODA home directory "C:\ProgramData\BOINC/projects/boinc.loda-lang.org_loda\"
2023-10-04 19:28:18|INFO |Checking environment
2023-10-04 19:28:18|WARN |Setting environment variable: COMSPEC=C:\WINDOWS\system32\cmd.exe
2023-10-04 19:28:18|WARN |Setting environment variable: SYSTEMROOT=C:\WINDOWS
2023-10-04 19:28:18|WARN |Setting environment variable: PATH=C:\WINDOWS\system32;C:\WINDOWS\system32\WindowsPowerShell\v1.0;C:\Program Files\Git\cmd;C:\ProgramData\BOINC/projects/boinc.loda-lang.org_loda\git\cmd;C:\Program Files\Git\usr\bin;C:\ProgramData\BOINC/projects/boinc.loda-lang.org_loda\git\usr\bin
2023-10-04 19:28:18|WARN |Setting environment variable: TMP=C:\ProgramData\BOINC/projects/boinc.loda-lang.org_loda\
2023-10-04 19:28:18|WARN |Setting environment variable: TEMP=C:\ProgramData\BOINC/projects/boinc.loda-lang.org_loda\
2023-10-04 19:28:18|INFO |Loading sequences from the OEIS index
2023-10-04 19:28:42|INFO |Loaded 353315/365669 sequences in 23.98s
2023-10-04 19:28:48|INFO |Initialized 6 matchers (ignoring 150140 sequences)
2023-10-04 19:28:49|INFO |Initialized 4 generators (profile: hard, overwrite: none)
2023-10-04 19:28:49|INFO |Mining programs in client mode (extended validation mode)
2023-10-04 19:29:05|INFO |Processed 4 programs, 0.7%
2023-10-04 19:29:42|INFO |Processed 437 programs, 1.2%
2023-10-04 19:30:07|ERROR|Caught exception during mining
2023-10-04 21:28:42|INFO |Initiating shutdown
```

**Comment by ckrause** [comment]
See also https://boinc.loda-lang.org/loda/forum_thread.php?id=75&postid=729

**Comment by ckrause** [comment]
Fixes are available in v23.10.8, but we need to check if we missed cases.
